src/GafferScene/EvaluateLightLinks.cpp

```diff
@@ -114,9 +114,13 @@ void EvaluateLightLinks::hashAttributes( const ScenePath &path, const Gaffer::Co
 	{
 		h.append( SetAlgo::setExpressionHash( illuminationExpressionData->readable(), inPlug() ) );
 	}
-	else
+	else if( path.size() == 1 )
 	{
-		h.append( inPlug()->setHash( m_defaultLightSetName ) );
+		const MurmurHash &defaultLightSetHash = inPlug()->setHash( m_defaultLightSetName );
+		if( defaultLightSetHash != inPlug()->setHash( "__lights" ) )
```

Comment:
We should preconstruct the InternedString for "__lights", and at the same time move `g_defaultLightSetName` internally as `g_defaultLightSetName` (no need for it to be a member).

Also, `setHash()` has the overhead of creating a temporary context internally. Since we're calling it twice, we should instead make our own `SetScope` and use it for both queries.